contracts/Pool.sol

```diff
@@ -20,6 +20,8 @@ contract MolochPool {
 
     bool locked; // prevent re-entrancy
 
+    uint256 constant MAX_NUMBER_OF_SHARES = 10**30; // maximum number of shares that can be minted
```

Comment:
I guess you added this to avoid overflows, right? Can you explain why it does?

Reply:
Yes, this was added to avoid overflows. Obviously, it will not work for tokens with very high supply, but since the Moloch contract was written with wETH in mind (and isn't approved for an arbitrary ERC-20 token), this seemed good enough for now.
The number `10**30` is, of course, debatable and (despite the current restriction to wETH) already geared towards a more token-agnostic approach. Roughly, the rationale for this number was: Many tokens have 18 decimals which often don't have much value. That leaves us with roughly 60 decimal digits, half of which I wanted to allocate for the shares, and the other half can be used by the token...